web/vtadmin/src/components/routes/tablets/PingDialog.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from 'react';
+import { usePingTablet } from '../../../hooks/api';
+import { Transition } from '@headlessui/react';
+import Dialog from '../../dialog/Dialog';
+import { Icon, Icons } from '../../Icon';
+
+const PingDialog: React.FC<{ alias: string; clusterID?: string; isOpen: boolean; onClose: () => void }> = ({
+    alias,
+    clusterID,
+    isOpen,
+    onClose,
+}) => {
+    // Mount content as separate component inside Dialog so internal queries are not executed unless dialog is open.
+    const PingContent: React.FC<{ isOpen: boolean }> = ({ isOpen }) => {
+        const { data: pingResponse, isLoading, isError, error, refetch } = usePingTablet({ alias, clusterID }, { enabled: false });
+        const [loading, setLoading] = useState(true)
+        useEffect(() => {
+            let timeout: NodeJS.Timeout
+            if (isOpen) {
+                setLoading(true)
+                timeout = setTimeout(() => {
+                    refetch();
+                    setLoading(isLoading)
+                }, 300);
+            }
+            return () => timeout && clearTimeout(timeout)
+        }, [isOpen, isLoading, refetch]);
```

Comment:
This looks great! Gating this on `isOpen` with `useEffect` makes a ton of sense.

I think shadowing the `isOpen` prop is unnecessary here (and maybe a little confusing). Now that we're manually fetching based on `isOpen`, I don't think we need the inner `PingContent` component + instead can flatten it out into `PingDialog` orrrr pull `PingContent` out into a separate component entirely (and add `alias` and `clusterID` as additional props). 

I don't think the extra function component definitions will affect reconciliation or performance in any meaningful way, but it's an opportunity to simplify a lil bit. Both would avoid the need to shadow the variable.

Reply:
I left isOpen there so as to avoid:
```
React Hook useEffect has a missing dependency: 'isOpen'. Either include it or remove the dependency array.eslintreact-hooks/exhaustive-deps
```
but can add the //eslint disable line to avoid. I'm leaving isOpen there to make sure `useEffect` is only called when isOpen changes 😄  and flattening it out to make it simpler.

Reply:
With those changes, can confirm the ping request still runs only once :-)
![pingtimes](https://user-images.githubusercontent.com/31225471/144932540-696802e7-b38f-477e-9ed5-9d14ab931066.gif)